lib/platform/linux-i2c.c

```diff
@@ -228,41 +244,134 @@ static uint8_t i2c_gas_cap_get(struct switchtec_dev *dev)
 		return (rx_buf[0] & GAS_TWI_MODE_MASK);
 }
 
-/* One I2C transaction can write a maximum of 28 bytes */
-#define I2C_MAX_WRITE 28
+/* One I2C transaction can write a maximum of 26 bytes */
+#define I2C_MAX_WRITE 24
 
-static void i2c_gas_write(struct switchtec_dev *dev, void __gas *dest,
-			  const void *src, size_t n)
+static uint8_t i2c_gas_data_write(struct switchtec_dev *dev, void __gas *dest,
+				  const void *src, size_t n, uint8_t tag)
 {
 	int ret;
 	struct switchtec_i2c *idev = to_switchtec_i2c(dev);
 
+	struct i2c_msg msg;
+	struct i2c_rdwr_ioctl_data wdata = {
+		.msgs = &msg,
+		.nmsgs = 1,
+	};
+
 	struct {
-		uint32_t gas_addr;
+		uint8_t command_code;
+		uint8_t byte_count;
+		uint8_t tag;
+		uint8_t offset[4];
```

Comment:
Why is this not a uint32_t? If it's a uint32_t, we don't need to memcpy below and can just do an assign (much nicer).

Reply:
If we use uint32_t, a uint8_t will be inserted between tag and offset because of the data alinement. When we send this data to I2C bus, it will be an incorrect I2C data flow for GAS TWI.

Reply:
Use "_attribute__((packed))" instead to avoid the padding. It will be clearer.

The compiler will also handle the unaligned access for you.